modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -403,7 +405,39 @@ object CrossSources {
     fromInputs ++ fromSources ++ fromSourcesWithRequirements
   }
 
-  private def resolveInputsFromSources(sources: Seq[Positioned[os.Path]], enableMarkdown: Boolean) =
+  // TODO: reuse existing one? e.g. scala.cli.commands.shared.SharedOptions.coursierCache
+  lazy val fileCache: FileCache[coursier.util.Task] = FileCache()
```

Comment:
Yup, I think it's accurate that `SharedOptions.coursierCache` should be used.

Reply:
@Gedochao but `scala.cli.commands.shared.SharedOptions.coursierCache` is in `cli` module, while we're in `build` here - so that definition is inaccessible (cli doesn't depend on build)

Reply:
That's why we haven't picked it up and left a todo

Reply:
...well, yes, but `cli` depends on `build`.
so what I mean here is that the way to go would be to pass it all the way from `cli` `to `CrossSources`.

Reply:
if you just leave a TODO, it's very likely to get forgotten.

Reply:
if you don't have time to fix it, please at least create a ticket for the follow-up and link it back in the code.

Reply:
@Gedochao, I think I understand what you mean by passing a download handler from the external. Could you please review the recent change?

Reply:
Yup, looks good!